lib/logic/models/playlist.dart

```diff
@@ -108,11 +108,12 @@ class Playlist extends PersistentQueue with DuplicatingSongOriginMixin implement
   }
 
   factory Playlist.fromMap(Map<String, dynamic> map) {
```

Comment:
this code could be code-generated, we should come up with a more reliable solution here

Reply:
Agreed, that would be nice, we should do that but in the mean time I think it is still valuable to have this fix in the mean time.

As discussed in #127, I also want the whole content scanning procedure to be more robust.

Reply:
I think here it's valuable to design a long-running solution from the start, because the moment this changes to codegen (which it likely will at some point), we might lose this part of logic, or will need to spend time re-writing it to something better

And from the architectural standpoint, it's not good to have error-reporting logic directly inside methods that implement model parsing

Reply:
I'm OK with fixing it like this right now, but we should create an issue to properly mark `dateModified` as being nullable and properly handle it on the business-logic level

Reply:
Ok, crated #157 to track it.